src/Sarif/Core/FileLocation.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public partial class FileLocation
```

Comment:
>FileLocation [](start = 25, length = 12)

Oh, right! All of our object model classes are `partial`. So, if you accept my proposed redesign in `TryGetUri`, you could put that method here, rather than making it an extension method.

Reply:
right! This helper makes sense right on the file location class...

---
In reply to: [205979970](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205979970) [](ancestors = 205979970)